components/GridView.vue

```diff
@@ -0,0 +1,216 @@
+<template>
+  <div>
+    <div ref="grid" class="grid">
+      <template v-for="card in cards">
+        <div
+          v-if="didShowMore && Array.isArray(card)"
+          :key="card.sortKey"
+          class="buddy"
+        >
+          <component
+            :is="components[_card.type]"
+            v-for="_card in card"
+            :key="_card.sortKey"
+            ref="card"
+            :card="_card"
+          />
+        </div>
+        <component
+          :is="components[card.type]"
+          v-else
+          :key="card.sortKey"
+          ref="card"
+          :card="card"
+          :columns="columns"
+        />
+      </template>
+      <component
+        :is="components[card.type]"
+        v-for="card in unusedCards.slice(0, emptySlots)"
+        :key="card.sortKey"
+        :card="card"
+        :columns="columns"
+      />
+    </div>
+
+    <div v-if="!loading" class="show-more">
+      <Coopknapp
+        v-if="!didShowMore && isMobile"
+        cta="Visa mer"
+        :disabled="loading"
+        @click="loadFull"
+      />
+    </div>
+    <div v-else-if="isMobile" class="show-more, loading">
+      <div class="lds-ellipsis">
+        <div />
+        <div />
+        <div />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import ProductCard from './ProductCard.vue'
+import InfoCard from './InfoCard.vue'
+import AdCard from './AdCard.vue'
+import SkeletonCard from './SkeletonCard.vue'
+import Coopknapp from './Coopknapp.vue'
+import { CardTypes } from '~/types/Card'
+import event from '@/event'
+
+export default Vue.extend({
+  components: {
+    Coopknapp
+  },
+  props: {
+    value: {
+      // cardWidth
+      type: Number,
+      required: true
+    },
+    columns: {
+      type: Number,
+      required: true
+    },
+    loading: Boolean,
+    didShowMore: Boolean,
+    isMobile: Boolean,
+    width: {
+      type: Number,
+      required: true
+    },
+    cards: {
+      type: Array,
+      required: true
+    },
+    unusedCards: {
+      type: Array,
+      required: true
+    },
+    emptySlots: {
+      type: Number,
+      required: true
+    },
+    loadFull: {
+      type: Function,
+      required: true
+    }
+  },
+  data() {
+    return {
+      components: {
+        [CardTypes.PRODUCT]: ProductCard,
+        [CardTypes.SKELETON]: SkeletonCard,
+        [CardTypes.INFO]: InfoCard,
+        [CardTypes.AD]: AdCard
+      }
+    }
+  },
+  watch: {
+    cards() {
+      this.setCardWidth()
+    }
+  },
+  mounted() {
+    this.setCardWidth()
+  },
+  methods: {
+    setCardWidth() {
+      if (this.$refs.card) {
+        event('card-width')
+        this.$emit('input', this.$refs.card[0].$el.clientWidth)
```

Comment:
I don't think we need to check card width, we can use the initial css value, and it will give us the same result.

@benjick How do we share values between script and style? It seems like it needs to be set once in the style and once in the script.